compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -90,11 +90,63 @@ struct LinalgCPUVectorLoweringPassOptions : LinalgVectorLoweringPassOptions {
 // Default allocation functions for CPU backend
 //===---------------------------------------------------------------------===//
 
+static std::optional<Value> hoistStaticallyBoundAllocations(
+    OpBuilder &builder, func::FuncOp funcOp, Location loc,
+    MemRefType memRefType, ValueRange dynamicSizes, unsigned alignment) {
+  int index = 0;
+
+  SmallVector<int64_t> staticShape;
+  SmallVector<OpFoldResult> subviewSizes;
+  staticShape.reserve(memRefType.getRank());
+  subviewSizes.reserve(memRefType.getRank());
+
+  for (auto dimSize : memRefType.getShape()) {
+    if (!ShapedType::isDynamic(dimSize)) {
+      staticShape.push_back(dimSize);
+      subviewSizes.push_back(builder.getIndexAttr(dimSize));
+      continue;
+    }
+    Value dynamicSize = dynamicSizes[index++];
+    auto ub = linalg::getConstantUpperBoundForIndex(dynamicSize);
+    if (failed(ub)) {
+      return std::nullopt;
+    }
+    staticShape.push_back(ub.value());
+    subviewSizes.push_back(dynamicSize);
+  }
+  SmallVector<OpFoldResult> offsets(memRefType.getRank(),
+                                    builder.getIndexAttr(0));
+  SmallVector<OpFoldResult> strides(memRefType.getRank(),
+                                    builder.getIndexAttr(1));
+
+  Value allocation;
+  {
+    OpBuilder::InsertionGuard g(builder);
+    builder.setInsertionPointToStart(&funcOp.getBody().front());
+    auto allocationType =
+        MemRefType::get(staticShape, memRefType.getElementType());
+    allocation = builder.create<memref::AllocaOp>(
+        loc, allocationType, builder.getI64IntegerAttr(alignment));
+  }
+
+  Value subviewOp = builder.create<memref::SubViewOp>(loc, allocation, offsets,
```

Comment:
Clarification: any subviews can be subsequently written into, right? If so, all this looks okay.

Reply:
yeah, you can do whatever with it. Its just an allocation.